satpy/composites/__init__.py

```diff
@@ -815,11 +815,12 @@ def __call__(self, projectables, nonprojectables=None, **attrs):
                 data['time'] = [time]
 
         new_attrs = combine_metadata(*projectables)
-        # remove metadata that shouldn't make sense in a composite
-        new_attrs["wavelength"] = None
-        new_attrs.pop("units", None)
-        new_attrs.pop('calibration', None)
-        new_attrs.pop('modifiers', None)
+        # remove metadata that shouldn't make sense in a multiband composite
+        if len(projectables) > 1:
+            new_attrs["wavelength"] = None
+            new_attrs.pop("units", None)
+            new_attrs.pop('calibration', None)
+            new_attrs.pop('modifiers', None)
```

Comment:
Doing this is risky. There are things like the DNB enhancements or any other single band composite where accidentally keeping any one of these would be incorrect. Especially wavelength. *Yes* the composite could/should be removing these but I think we've been depending on this being part of the base compositor. I could also see this being separated where BaseCompositor has all of the above functionality, but GenericCompositor (being made for images - L, RGB, etc) could remove these attributes because they don't make sense.

What composites do you have that you need to keep this information?

Reply:
IR. It is important that we keep the units so that it can be converted to the right unit later.

Reply:
Let me check if I can revert this

Reply:
What "IR"? If it is using GenericCompositor then it is an image and should be considered unitless, right? The subclass compositor could also re-add units although I'm not sure that is great either.

Reply:
That's the thing with tiffs like these: the values in the file form an image be must be revertable to physical quantities

Reply:
so it's both an image and a physical dataset